cli/bptest/metadata_lint.go

```diff
@@ -0,0 +1,126 @@
+package bptest
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+
+	"gopkg.in/yaml.v2"
+)
+
+// BlueprintConnection structure to match metadata YAML
+type BlueprintConnection struct {
+	Source  string `yaml:"source"`
+	Version string `yaml:"version"`
+}
+
+// BlueprintVariable structure to match metadata YAML
+type BlueprintVariable struct {
+	Connections []BlueprintConnection `yaml:"connections"`
+}
+
+// BlueprintInterface structure to match metadata YAML
+type BlueprintInterface struct {
+	Variables []BlueprintVariable `yaml:"variables"`
+}
+
+// BlueprintMetadata structure for entire metadata YAML
+type BlueprintMetadata struct {
+	Spec struct {
+		Interfaces []BlueprintInterface `yaml:"interfaces"`
+	} `yaml:"spec"`
+}
+
+// validateMetadataFile validates a metadata.yaml file by checking blueprint references and versions
+func validateMetadataFile(metadataPath string) error {
+	fmt.Printf("Validating: %s\n", metadataPath)
+
+	// Read the metadata file
+	content, err := ioutil.ReadFile(metadataPath)
+	if err != nil {
+		return fmt.Errorf("error reading metadata file %s: %v", metadataPath, err)
+	}
+
+	// Unmarshal the YAML content into the BlueprintMetadata struct
+	var metadata BlueprintMetadata
+	if err := yaml.Unmarshal(content, &metadata); err != nil {
+		return fmt.Errorf("error parsing metadata file %s: %v", metadataPath, err)
+	}
+
+	// Walk through the BlueprintInterface -> BlueprintVariable -> BlueprintConnection
+	for _, blueprintInterface := range metadata.Spec.Interfaces {
+		for _, blueprintVariable := range blueprintInterface.Variables {
+			for _, connection := range blueprintVariable.Connections {
+				// Validate blueprint source and version
+				if err := validateBlueprintSource(connection.Source); err != nil {
+					return err
+				}
+				if err := validateBlueprintVersion(connection.Version, connection.Source); err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+var httpGetFunc = http.Get
```

Comment:
We should avoid globals if possible. For testing I would recommend using [httptest](https://pkg.go.dev/net/http/httptest).

Reply:
Removed